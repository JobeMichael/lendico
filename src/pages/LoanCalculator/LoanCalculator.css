.fields-wrapper {
  display: flex;
  justify-content: center;
  align-items: center;
  box-shadow: inset 0 0 7px #ccc;
  padding: 40px 0;
  border-radius: 3px;
}

.monthly-rate {
  box-shadow: inset 0 0 7px #ccc;
  padding: 20px 0;
  border-radius: 3px;
  min-height: 300px;
  display: flex;
  align-items: center;
  justify-content: center;
}
