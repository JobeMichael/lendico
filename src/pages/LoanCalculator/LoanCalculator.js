import React, { useState } from "react";
import { useForm } from "react-hook-form";
import MonthlyRate from "../../components/MonthlyRate/MonthlyRate";
import Button from "../../components/UI/Button/Button";
import Input from "../../components/UI/Input/Input";
import Spinner from "../../components/UI/Spinner/Spinner";
import validation from "../../helper/validation";
import loanCalculator from "../../services/loanCalculator";
import "./LoanCalculator.css";

const LoanCalculator = () => {
  const [monthlyInstallment, setMonthlyInstallment] = useState("");
  const [loading, setLoading] = useState(false);

  const { register, errors, handleSubmit } = useForm({
    validateCriteriaMode: "all",
  });

  const onSubmit = async (formValues) => {
    setLoading(true);
    const { amount, duration } = formValues;
    const { data } = await loanCalculator.postLoanCalculator("loanCalculator", {
      amount,
      duration,
    });
    setMonthlyInstallment(data?.monthlyInstallment || "");
    setLoading(false);
  };

  return (
    <>
      <form onSubmit={handleSubmit(onSubmit)}>
        <div className="fields-wrapper">
          <Input
            name="amount"
            label="Amount"
            type="number"
            inputRef={register(validation.amount)}
            errors={errors}
          />
          <Input
            name="duration"
            label="Duration"
            type="number"
            inputRef={register(validation.duration)}
            errors={errors}
          />
          <Button disabled={false}>OK</Button>
        </div>
      </form>
      <div className="monthly-rate">
        {loading ? (
          <Spinner />
        ) : (
          <MonthlyRate monthlyRate={monthlyInstallment} />
        )}
      </div>
    </>
  );
};

export default LoanCalculator;
