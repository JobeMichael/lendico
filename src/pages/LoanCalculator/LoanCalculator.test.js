import { render } from "@testing-library/react";
import MutationObserver from "mutation-observer";
import React from "react";
import LoanCalculator from "./LoanCalculator";
global.MutationObserver = MutationObserver;

describe("rendering", () => {
  test("renders LoanCalculator correctly", () => {
    const { asFragment } = render(<LoanCalculator />);

    expect(asFragment()).toMatchSnapshot();
  });
});
