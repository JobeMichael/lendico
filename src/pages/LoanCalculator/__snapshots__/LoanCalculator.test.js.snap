// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering renders LoanCalculator correctly 1`] = `
<DocumentFragment>
  <form>
    <div
      class="fields-wrapper"
    >
      <div
        class="field"
      >
        <label
          class="label"
        >
          Amount : 
        </label>
        <div
          class="control"
        >
          <input
            autocomplete="off"
            class="input-element"
            name="amount"
            type="number"
          />
        </div>
      </div>
      <div
        class="field"
      >
        <label
          class="label"
        >
          Duration : 
        </label>
        <div
          class="control"
        >
          <input
            autocomplete="off"
            class="input-element"
            name="duration"
            type="number"
          />
        </div>
      </div>
      <button
        class="button"
        type="submit"
      >
        OK
      </button>
    </div>
  </form>
  <div
    class="monthly-rate"
  >
    <p
      data-testid="infoText"
    >
      Enter Amount and duration to calculate the installment
    </p>
  </div>
</DocumentFragment>
`;
